astropy/convolution/tests/test_convolve_models.py

```diff
@@ -0,0 +1,94 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import math
+import numpy as np
+from ..convolve import convolve, convolve_models
+from ...modeling import models, fitting
+from ...tests.helper import pytest
+from ...utils.misc import NumpyRNGContext
+from numpy.testing import assert_allclose, assert_almost_equal
+
+try:
+    import scipy
+except ImportError:
+    HAS_SCIPY = False
+else:
+    HAS_SCIPY = True
+
+class TestConvolve1DModels(object):
+    @pytest.mark.parametrize('mode', ['convolve_fft', 'convolve'])
+    @pytest.mark.skipif('not HAS_SCIPY')
+    def test_against_scipy(self, mode):
+        from scipy.signal import fftconvolve
+
+        kernel = models.Gaussian1D(1, 0, 1)
+        model = models.Gaussian1D(1, 0, 1)
+        model_conv = convolve_models(model, kernel, mode=mode)
+        x = np.arange(-5, 6)
+        ans = fftconvolve(kernel(x), model(x), mode='same')
```

Comment:
it would be better to use astropy's convolution here to test self-consistency, but add the "is astropy consistent with scipy? numpy?" tests in `test_convolve` separately.